lib/ipam/ipam_block_reader_writer.go

```diff
@@ -276,29 +273,27 @@ func (rw blockReaderWriter) releaseBlockAffinity(host string, blockCIDR cnet.IPN
 
 		// We've removed / updated the block, so update the host config
 		// to remove the CIDR.
-		err = rw.client.Backend.Delete(&model.KVPair{
-			Key: model.BlockAffinityKey{Host: host, CIDR: b.CIDR},
-		})
+		_, err = rw.client.Delete(context.Background(), model.BlockAffinityKey{Host: host, CIDR: b.CIDR}, "")
 		if err != nil {
 			// Return the error unless the affinity didn't exist.
-			if _, ok := err.(errors.ErrorResourceDoesNotExist); !ok {
+			if _, ok := err.(cerrors.ErrorResourceDoesNotExist); !ok {
 				log.Errorf("Error deleting block affinity: %s", err)
 				return err
 			}
 		}
 		return nil
 
 	}
-	return goerrors.New("Max retries hit")
+	return errors.New("Max retries hit")
 }
 
 // withinConfiguredPools returns true if the given IP is within a configured
 // Calico pool, and false otherwise.
 func (rw blockReaderWriter) withinConfiguredPools(ip cnet.IP) bool {
-	allPools, _ := rw.client.IPPools().List(api.IPPoolMetadata{})
-	for _, p := range allPools.Items {
+	allPools, _ := rw.pools.GetEnabledPools(ip.Version())
```

Comment:
same here, it's not not allPools anymore

Reply:
kk, tweaking.